sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetQuerySuite.scala

```diff
@@ -879,6 +879,18 @@ class ParquetQuerySuite extends QueryTest with ParquetTest with SharedSQLContext
       }
     }
   }
+
+  test("SPARK-24230: filter row group using dictionary") {
+    withSQLConf(("parquet.filter.dictionary.enabled", "true")) {
```

Comment:
Is this a valid way to control this configuration? It seems to pass with `false`, too.

Reply:
This is the default, so it is possible that it isn't getting passed to Parquet correctly. I can debug it at some point to find out why it passes with `false`. I did make sure that the test case fails without the fix, so we know it should be correctly using dictionary filtering.

Reply:
If there is someone who leads this correctly, it's you, @rdblue . :)

I knew that this is the default of parquet. With the patch of `SpecificParquetRecordReaderBase.java` or not, we should not add `no-op` invalid conf line like `withSQLConf` here. It's misleading for the whole Spark community for the future. Please debug and add the correct test case here.

Reply:
We are unable to pass the parquet-specific parameter through `withSQLConf`. Below shows the way to pass the parquet option.  

```Scala
    withTable("t1") {
      spark.createDataFrame((0 until 100).map(i => ((i * 2) % 20, s"data-$i"))).write
        .option("parquet.filter.dictionary.enabled", false).saveAsTable("t1")
      checkAnswer(sql("SELECT _2 FROM t1 WHERE t1._1 = 5"), Seq.empty)
    }
```

Could you help investigate why we still hit the error [without the fix] when we set `parquet.filter.dictionary.enabled` to `false`?

Reply:
using SQLConf is OK, Spark will put all SQL configs to Hadoop conf, which will be accessed by parquet writer at the executor side.

I'm also curious about why turning `parquet.filter.dictionary.enabled` off can't avoid this bug.

Reply:
Looks like the problem is a bug in Parquet. It is using the [stats property instead of the dictionary property](https://github.com/apache/parquet-mr/blob/8bbc6cb95fd9b4b9e86c924ca1e40fd555ecac1d/parquet-hadoop/src/main/java/org/apache/parquet/HadoopReadOptions.java#L83). This is minor because there is almost no reason to turn either one off now that we've built more confidence in the filters.